Golems cannot turn into Plasmamen

Jacquerel opened the pull request:
## About The Pull Request

Fixes #79155 
Also cleans up the code in plasma river "lava" a bit

The plasma river code was a little crusty and confusing and made some interesting decisions.
To start with it would do _two_ burn damage per tick to people entering it.
Then _15 toxic and 25 more burn damage_ to specifically humans?
This seems... unintended? So now it does the same amount of damage equally to anyone regardless of if they are human or not.

In terms of the "turning dead people into dead plasmamen" behaviour we will now:
- Not attempt to transform any limbs which cannot be dismembered, with the assumption that those aren't meant to be replaceable.
- Not attempt to transform anyone with the undismemberable trait, with the assumption that their body is supposed to be immutable.
- Not attempt to transform anyone with the new "don't transform me, plasma river" trait which replaces an explicit species check for being an android or plasmaman (already).

I also simply made golems immune to the plasma river _anyway_, because people expect them to be able to wade into lava and the fact that this is "cold lava" probably shouldn't impede that.

I also made golems unable to become husks because... none of the three occasions where this could happen really apply and also if they get husked they turn into gross-looking humans instead of golems until fixed, which is weird.

Finally I did some more fucking around with husking because it's visually weird.
I added a new flag to bodyparts (applied to robotic and skeletal parts) which means that they will never be replaced with a husk icon. It might be preferable to instead make custom husk sprites for these limbs? But half of them are on species which cannot be husked in the first place so this would only appear when they are transplanted onto a species which can.

## Changelog

:cl:
fix: The plasma river is about as deadly for animals as it is for humans.
fix: Golems can now wade in the plasma river unscathed.
fix: Undismemberable limbs will no longer be dismembered by the plasma river.
balance: Golems and plasmamen cannot become husked.
image: Robotic and Skeletal parts will remain distinct while the rest of the body is husked.
/:cl:

Jacquerel committed:
Golems cannot become Plasmamen

Jacquerel committed:
Just let them wade in there

Jacquerel committed:
minor tweak

MrMelbert commented:
Also the code beyond the parent call here is dead code 

![image](https://github.com/tgstation/tgstation/assets/51863163/0099aef7-8635-4223-ab15-3d77f403e5a3)

...Because it doesn't return a truthy value

![image](https://github.com/tgstation/tgstation/assets/51863163/4998246a-c500-4a4d-960e-bf786bc728e8)

san7890 commented:
MMMiracles Snowdin code comes back to haunt us i hate it here

Overuseddart commented:
Won’t this kill all the fauna that wander into it?

Jacquerel commented:
> Won’t this kill all the fauna that wander into it? Cause they will just stay in it and not get out

Most icebox fauna is already immune to plasma river damage

Jacquerel committed:
Skeletons and Plasmamen are also unhuskable
Becoming unhuskable while husked (for instance if you are turned into a plasmaman) removes the husked status
Some bodyparts will never change into a husked overlay

Jacquerel commented:
One day some brave soul will make limbs be individually capable of catching fire and remove that code from the plasmaman species but that day isn't today and that soul isn't me.

GPeckman commented:
I kinda feel like androids should also have the unhuskable trait. Or at least have species-specific methods for unhusking that don't use synthflesh. Using synthflesh to unhusk a robot is just weird.

Jacquerel commented:
This is failing consistently enough, and on icebox, that maybe it actually related to my change but fuck if I know how

Jacquerel committed:
Merge branch 'master' into cool-lava

Jacquerel committed:
Merge branch 'master' into cool-lava
